packages/remix-react/components.tsx

```diff
@@ -605,12 +605,22 @@ export type ScriptProps = Omit<
  * @see https://remix.run/components/scripts
  */
 export function Scripts(props: ScriptProps) {
-  let { manifest, serverHandoffString, abortDelay, serializeError } =
+  let { manifest, serverHandoffString, abortDelay, serializeError, isSpaMode } =
     useRemixContext();
   let { router, static: isStatic, staticContext } = useDataRouterContext();
-  let { matches } = useDataRouterStateContext();
+  let { matches: dontUseTheseMatches } = useDataRouterStateContext();
   let navigation = useNavigation();
 
+  // Use these `matches` instead :)
+  // In SPA Mode we only want to import root on the critical path, since we
+  // want the generated HTML file to be able to be loaded at non-/ paths as
+  // well.  This lets the router handle initial match loads via lazy().
+  //
+  // TODO: If users only ever intend to serve the `/` path from index.html and
+  // thus they want to include all initial matches here...should we give them
+  // a config to do so?
+  let matches = isSpaMode ? [dontUseTheseMatches[0]] : dontUseTheseMatches;
```

Comment:
@mjackson @ryanflorence Thoughts?  I like the ability to load any URL (`/`, `/path`, `/path/child`) via index.html (like you can do in vite/RR) - but in order to do so we have to slightly de-optimize by not choosing which route to pre-load.  If a user wanted to only ever support landing on index.html via `/`, we could optimize and include the sub-matches here.

Reply:
We agreed we do want to hydrate at any path